plugins/kubernetes.yaml

```diff
@@ -1,186 +1,191 @@
-        severity:
-          parse_from: severity
-          mapping:
-            debug: d
-            info: i
-            warning: w
-            error: e
-            critical: c
-        timestamp:
-          parse_from: timestamp
-          layout: '%m%d %H:%M:%S.%s'
-        output: add_kublet_metadata
-
-      - id: add_kublet_metadata
-        type: metadata
-        labels:
-          log_name: 'kubernetes.kubelet'
-        output: {{ .output }}
+version: 0.0.5
+title: Kubernetes
+description: Log parser for Kubernetes
+parameters:
+  container_log_path:
+    label: Containers Log Path
+    description: Kubernetes Containers Log Path
+    type: string
+    required: true
+  kubelet_journald_log_path:
+    label: Kublet Journald Log Path
+    description: 'Kubernetes Kublet Journald Log path. It will read from /run/journal or /var/log/journal if this parameter is omitted'
+    type: string
+  start_at:
+    label: Start At
+    description: "Start reading file from 'beginning' or 'end'"
+    type: enum
+    valid_values:
+     - beginning
+     - end
+    default: end
+pipeline:
+  # {{ if .container_log_path }}
+  - id: container_reader
+    type: file_input
+    include:
+      - {{ .container_log_path }}
+    # {{ if .start_at }}
+    start_at: {{ .start_at }}
+    # {{ end }}
+    file_path_field: log_name
+    write_to: log
+    output: container_json_parser
+
+  - id: container_json_parser
+    type: json_parser
+    parse_from: log
+    output: nested_json_router
+
+  - id: nested_json_router
+    type: router
+    routes:
+      - output: nested_json_parser
+        expr: $record.log matches '^{.*}$'
+      - output: container_regex_parser
+        expr: true
+
+  - id: nested_json_parser
+    type: json_parser
+    parse_from: $.log
+    output: container_regex_parser
+
+  - id: container_regex_parser
+    type: regex_parser
+    parse_from: log_name
+    regex: '\/var\/log\/containers\/(?P<pod_name>[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)_(?P<namespace>[^_]+)_(?P<container_name>.+)-(?P<container_id>[a-z0-9]{64})\.log'
+    severity:
+      parse_from: stream
+      preserve: true
+      mapping:
+        error:
+          - stderr
+        info:
+          - stdout
+    timestamp:
+      parse_from: time
+      layout: '%Y-%m-%dT%H:%M:%S.%sZ'
+    output: log_parse_router
+
+  - id: log_parse_router
+    type: router
+    routes:
+      - output: standard_regex_parser
+        expr: '$record.log matches "^\\w\\d{4}"'
+      - output: add_kubernetes_metadata
+        expr: true
+
+  - id: standard_regex_parser
+    type: regex_parser
+    parse_from: log
+    regex: '(?P<severity>\w)(?P<timestamp>\d{4} \d{2}:\d{2}:\d{2}.\d+)\s+(?P<pid>\d+)\s+(?P<source>[^ \]]+)\] (?P<message>.*)'
+    severity:
+      parse_from: severity
+      mapping:
+        debug:
+          - d
+        info:
+          - i
+        warning:
+          - w
+        error:
+          - e
+        critical:
+          - c
```

Comment:
Since we are changing the file anyways I recommend we update this to the more compact style.
        debug: d
        info: i
        warning: w
        error: e
        critical: c